naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2762,14 +2839,34 @@ impl Parser {
         }
     }
 
-    pub fn parse<'a>(&mut self, source: &'a str) -> Result<ast::TranslationUnit<'a>, Error<'a>> {
+    pub fn parse<'a>(
+        &mut self,
+        source: &'a str,
+        options: &Options,
+    ) -> Result<ast::TranslationUnit<'a>, Error<'a>> {
         self.reset();
 
-        let mut lexer = Lexer::new(source);
+        let mut lexer = Lexer::new(source, options.parse_comments);
         let mut tu = ast::TranslationUnit::default();
         let mut enable_extensions = EnableExtensions::empty();
         let mut diagnostic_filters = DiagnosticFilterMap::new();
 
+        if options.parse_comments {
+            // Parse module comments.
+            let mut comments = Vec::new();
+
+            fn peek_any_next<'a>(lexer: &'a Lexer) -> (Token<'a>, Span) {
+                let mut cloned = lexer.clone();
+                let token = cloned.next_until(|_| true, false);
+                token
+            }
+            while let (Token::CommentDocModule(_), span) = peek_any_next(&lexer) {
+                comments.push(lexer.source.index(span));
```

Comment:
Doesn't `Token::CommentDocModule(_)` already contain the `&str`?

Reply:
Correct! as I'm peeking I can't use it though because the borrow checker is unhappy, I think I have to make a `next_module_comment` function for the lexer.

Reply:
Actually, it's not 100% trivial, changing this would either:

- require a `peek` before, then getting the comment if relevant.
- or use an `accumulate_comments` for module docs, then backtracking to get the last token which resulted in leaving that `accumulate_comments`.
- change `accumulate_comment` implementation to return the last `(token, rest)`.

Reply:
A similar method to `accumulate_doc_item_comments` would work and we wouldn't need to backtrack.